_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,289 @@
+
+  **Questions**:
+
+  1. Exercise 1 in Section 11.12 of (Quarteroni et al.)
+     <details><summary>Solutions</summary>
+     <p>
+     ...
+     </p>
+     </details>
+
+  2. Exercises 12.3,12.4, 12.7 in Section 12 of (S\"uli & Mayers)
+     <details><summary>Solution</summary>
+     <p>
+     ...
+     </p>
+     </details>
+
+  3. Consider the following method for solving \(y' = f(y)\):
+       
+     $$y_{n+1} = y_n + h*(theta*f(y_n) + (1-theta)*f(y_{n+1}))$$
+    
+     Assuming sufficient smoothness of \(y\) and \(f\), for what value of \(0 \leq\theta\leq 1\) is the truncation error the smallest? What does this mean about the accuracy of the method?
+     <details><summary>Solution</summary>
+     <p>
+     ...
+     </p>
+     </details>
+       
+
+  4. [Colab Notebook](https://colab.research.google.com/drive/1bNg-RzZoelB3w8AUQ6mefRQuN3AdrIqX)
+     <details><summary>Solution</summary>
+     <p>
+     ...
+     </p>
+     </details>
+
+**Notes**: Here is a [link](/assets/nodes_notes/week1.pdf) to our notes for the lesson. 
+
+<br />
+
+# 2 Numerical solution of ODEs - Part 2
+  **Motivation**: In the previous class we introduced some simple schemes to numerically solve ODEs. In this class we go through some more involved schemes and their convergence analysis. 
+
+  **Topics**:
+
+  1. Runge-Kutta methods
+  2. Multi-step methods
+  3. System of ODEs and absolute converge
+
+  **Required Reading**:
+
+  1. Runge-Kutta methods: Section 11.8 from (Quarteroni et al.) or Sections 12.{5,12} from (S\"uli & Mayers)
+  2. Multi-step methods: Sections 12.6-9 from (Quarteroni et al.) or Section 11.5-6 from (S\"uli & Mayers)
+  3. System of ODEs: Sections 12.10-11 from (Quarteroni et al.) or Sections 11.9-10 from (S\"uli & Mayers)
+
+  
+  **Optional Reading**:
+
+  1. [Prof. Trefethen's class ODEs and Nonlinear Dynamics 4.1](http://podcasts.ox.ac.uk/odes-and-nonlinear-dynamics-41)
+  2. Predictor-corrector methods: Section 11.7 from (Quarteroni et al.)
+  3. Richardson extrapolation: Section 16.4 from [Numerical Recipes](http://numerical.recipes/)
+  4. [Automatic Selection of Methods for Solving Stiff and Nonstiff Systems of Ordinary Differential Equations](https://epubs.siam.org/doi/pdf/10.1137/0904010?
+  
+  **Questions**:
+
+  1. Exercises 12.11, 12.12, 12.19 in Section 12 of (S\"uli & Mayers)
+     <details><summary>Solution</summary>
+     <p>
+     ...
+     </p>
+     </details>
+
+**Notes**: Here is a [link](/assets/nodes_notes/week2.pdf) to our notes for the lesson. 
+
+<br />
+
+# 3 ResNets
+  **Motivation**: The introduction of Residual Networks (ResNets) made possible to train very deep networks. In this section we study some residual architectures variants and their properties. We then look into how ResNets approximates ODEs and how this interpretation can motivate neural net architectures and new training approaches. 
```

Comment:
"The introduction of Residual Networks (ResNets) made *it* possible to train very deep networks. In this section, we study residual architecture variants and their properties. We then look into how ResNets approximate ODEs and how this interpretation can motivate new training approaches."

Why?

Reply:
added